### ISSUE TITLE
[android] eject production build vector-icons seem to be missing

### ISSUE THREAD

**Issue by aniltirola** [expo/expo#3480]
### Environment
platform: `android`
type of build: `ejected, release-build`
updates enabled: `false`

expo-cli --version: 2.10.0
node --version: v10.15.1

packages:
```
"expo": "^32.0.0",
"expokit": "32.0.6",
"react": "16.5.0",
"react-native": "https://github.com/expo/react-native/archive/sdk-32.0.0.tar.gz"
```

### Steps to Reproduce

lets think about this peace of code

```javascript
import React from 'react';
import { Text, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

export default class App extends React.Component {
  render() {
    return (
      <View style={{flex:1, justifyContent: 'center', alignItems: 'center'}}>
        <Text>Below you should see the green checkmark</Text>
        <Ionicons name="md-checkmark-circle" size={32} color="green" />
      </View>
    );
  }
}
```

### Expected Behavior

Should look like this:

![image](https://user-images.githubusercontent.com/1217678/52550726-973cad00-2dd9-11e9-9fc1-29da7e57d9db.png)

### Actual Behavior

- during development/debugging: working
- when building a signed apk with android-studio: green checkmark not showing

### Analysis
After building a signed apk, all needed vector-fonts are available in `app/build/intermediates/assets/prodKernel/release`

It seems that they are not correctly bundled into the apk during the build-process.

### Temporary workaround

1. eject
2. publish your experience
3. open Android Studio
4. build release apk
5. after build is complete go to folder `app/build/intermediates/assets/prodKernel/release`
6. click on every `*.ttf` - file and check which fonts are needed by your app; in the example you only need the file which contains the `Ionicons`
7. copy needed files to `android/app/src/main/assets`
8. rebuild release-apk

**Comment by gmaclennan** [comment]
I am seeing this same issue, and have reproduced it with a new repo from scratch with expo-cli v2.10.1 with the following steps:

### Environment
platform: `android`
type of build: `ejected, release-build`
updates enabled: `true`

expo-cli --version: 2.10.1
node --version: v8.15.0

### Steps to reproduce

```sh
expo init mapeo
# Selected tabs example
expo eject
expo publish --release-channel=production
cd android
./gradlew assembleProdKernelRelease
adb install app/build/outputs/apk/prodKernel/release/app-prodKernel-release.apk
```

Loading the app in the emulator the PNG icon and the font icons are missing:

![image](https://user-images.githubusercontent.com/290457/52571295-36c76300-2e0d-11e9-9731-b993ab2fee31.png)

Potentially relevant lines from `adb logcat`:

```txt
02-11 14:57:30.605  4205  4274 D b       : Using cached or embedded response.
02-11 14:57:30.605  4205  4274 D b       : Got cached OkHttp response for https://d1wp6m56sqw74a.cloudfront.net/%40digidem%2Fmapeo%2F1.0.0%2Fc7754be681d48621056fcd337b9d85c6-32.0.0-android.js
02-11 14:57:30.605  4205  4274 D b       : Have cached source file for https://d1wp6m56sqw74a.cloudfront.net/%40digidem%2Fmapeo%2F1.0.0%2Fc7754be681d48621056fcd337b9d85c6-32.0.0-android.js
02-11 14:57:30.606  4205  4205 E f       : java.lang.NullPointerException: Attempt to invoke virtual method 'host.exp.exponent.j host.exp.exponent.j.b(java.lang.String, java.lang.Object[])' on a null object reference
02-11 14:57:30.607  4205  4232 E GraphResponse: {HttpStatus: 400, errorCode: 100, subErrorCode: 33, errorType: GraphMethodException, errorMessage: Unsupported post request. Please read the Graph API documentation at https://developers.facebook.com/docs/graph-api}
02-11 14:57:30.667  4205  4268 D ReactNative: Initializing React Xplat Bridge.
02-11 14:57:30.668  4205  4268 D ReactNative: Initializing React Xplat Bridge before initializeBridge
02-11 14:57:30.673  4205  4268 D ReactNative: Initializing React Xplat Bridge after initializeBridge
02-11 14:57:30.673  4205  4268 D ReactNative: CatalystInstanceImpl.runJSBundle()
02-11 14:57:30.673  4205  4268 D CatalystInstanceImpl: Loading script from file
02-11 14:57:30.674  4205  4277 D ReactNative: ReactInstanceManager.setupReactContext()
02-11 14:57:30.674  4205  4277 D ReactNative: CatalystInstanceImpl.initialize()
02-11 14:57:30.676  4205  4277 D ReactNative: ReactInstanceManager.attachRootViewToInstance()
02-11 14:57:30.676  4205  4277 D SoLoader: About to load: libyoga.so
02-11 14:57:30.676  4205  4277 D SoLoader: libyoga.so not found on /data/data/com.mapeomobile/lib-main
02-11 14:57:30.676  4205  4277 D SoLoader: libyoga.so found on /data/app/com.mapeomobile-wZYWoABcQS-CCt4fsNNQIQ==/lib/x86
02-11 14:57:30.676  4205  4277 D SoLoader: Not resolving dependencies for libyoga.so
02-11 14:57:30.698  4205  4277 D SoLoader: Loaded: libyoga.so
02-11 14:57:30.704  4205  4277 W unknown:Fresco: Fresco has already been initialized! `Fresco.initialize(...)` should only be called 1 single time to avoid memory leaks!
02-11 14:57:30.704  4205  4277 D SoLoader: init exiting
02-11 14:57:30.704  4205  4277 W unknown:ImagePipelineFactory: ImagePipelineFactory has already been initialized! `ImagePipelineFactory.initialize(...)` should only be called once to avoid unexpected behavior.
02-11 14:57:30.813  4205  4276 D j       : WARNING: getPackageName called on ScopedContext
02-11 14:57:31.128  4205  4276 I ReactNativeJS: Running application "main" with appParams: {"initialProps":{"exp":{"initialUri":"exp250c0369fd594ab9b6addfee5bb6b1a8://","manifest":{"iconUrl":"https://d1wp6m56sqw74a.cloudfront.net/~assets/fa6577fecc0a7838f15a254577639984","releaseChannel":"production","isVerified":true,"hostUri":"exp.host/@digidem/mapeo?release-channel=production","splash":{"resizeMode":"contain","backgroundColor":"#ffffff","image":"./assets/images/splash.png","imageUrl":"https://d1wp6m56sqw74a.cloudfront.net/~assets/43ec0dcbe5a156bf9e650bb8c15e7af6"},"privacy":"public","commitTime":"2019-02-11T14:11:18.491Z","scheme":"exp250c0369fd594ab9b6addfee5bb6b1a8","slug":"mapeo","locales":{},"detach":{"androidExpoViewUrl":"https://s3.amazonaws.com/exp-exponent-view-code/android-v2.10.5-sdk32.0.0-cfdf42e8-f9b5-4ddd-84fd-2f2aacffe2c1.tar.gz","iosExpoViewUrl":"https://s3.amazonaws.com/exp-exponent-view-code/ios-v2.10.2-sdk32.0.0-c59c23cd-1d49-49d2-a633-d03f9d554cf6.tar.gz"},"orientation":"portrait","android":{"package":"com.mapeomobile","publishManifestPath":"android/app/src/main/assets/shell-app-manifest.json","publishBundlePath":"android/app/src/main/assets/shell-app.bundle"},"bundleUrl":"https://d1wp6m56sqw74a.cloudfront.net/%40digidem%2Fmapeo%2F1.0.0%2Fc7754be681d48621056fcd337b9d85c6-32.0.0-android.js","bundledAssets":["asset_3a2ba31570920eeb9b1d217cabe58315.ttf","asset_744ce60078c17d86006dd0edabcd59a7.ttf","asset_140c53a7643ea949007aa9a282153849.ttf","asset_6beba7e6834963f7f171d3bdd075c915.ttf","asset_b06871f281fee6b241d60582ae9369b9.ttf","asset_e20945d7c929279ef7a6f1db184a4470.ttf","asset_b2e0fc821c6886fb3940f85a3320003e.ttf","asset_d15c1216957060fac577af6151fb8cfe.ttf","asset_a37b0c01c0baf1888ca812cc0508f6e2.ttf","asset_73b8cff012825060b308d2162f31dbb2.ttf","asset_d2285965fe34b05465047401b8595dd0.ttf","asset_5cdf883b18a5651a29a4d1ef276d2457.ttf","asset_7d40544b395c5949f4646f5e150fe020.png","asset_cdd04e13d4ec83ff0cd13ec8dabdc341.png","asset_a132ecc4ba5c1517ff83c0fb321bc7fc.png","asset_0ea69b5077e7c4696db85dbcba75b0e1.png","asset_f5b790e2ac193b3d41015edb3551f9b8.png","asset_6165c9d7a2e729ba57b23dd93add5366.png","asset_45a8ce76165dcaec1c5e39be15478e53.png","asset_2d8deea13c41b52b04442d91fb31c2c5.png","asset_5c36108361d0a750fd25533356f80245.png","asset_af47141a24eb8e5751dfd034159479bd.png","asset_ad107d1ca027259ca50ce0f432293e6c.png","asset_a30cd76fa74ee5a0bab5aba37c101416.png","asset_49a79d66bdea2debf1832bf4d7aca127.ttf","asset_778ffc9fe8773a878e9c30a6304784de.png","asset_376d6a4c7f622917c39feb23671ef71d.png","asset_c79c3606a1cf168006ad3979763c7e0c.png","asset_02bc1fa7c0313217bde2d65ccbff40c9.png","asset_35ba0eaec5a4f5ed12ca16fabeae451d.png"],"updates":{"fallbackToCacheTimeout":0},"ios":{"supportsTablet":true,"publishManifestPath":"ios/mapeo/Supporting/shell-app-manifest.json","bundleIdentifier":"com.mapeomobile","publishBundlePath":"ios/mapeo/Supporting/shell-app.bundle"},"sdkVersion":"32.0.0","revisionId":"1.0.0-r.oe8dlD8rWx","publishedTime":"2019-02-11T14:11:18.397Z","name":"Mapeo","isDetached":true,"platforms":["ios","android"],"id":"@digidem/mapeo","loadedFromCache":true,"icon":"./assets/images/icon.png","version":"1.0.0"},"shell":true}},"rootTag":1}. __DEV__ === false, development-level warning are OFF, performance optimizations are ON
02-11 14:57:31.167  4205  4276 W ReactNativeJS: { [Error: Font.loadAsync unexpected exception: Font asset not found asset_b2e0fc821c6886fb3940f85a3320003e.ttf]
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   line: 20,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   column: 1565,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   sourceURL: '/data/user/0/com.mapeomobile/files/32.0.0/cached-bundle-experience-%40digidem%2Fmapeo2089336494-32.0.0',
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   framesToPop: 1,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   code: 'E_UNEXPECTED' }
02-11 14:57:31.250  4205  4205 I ReactNative: [GESTURE HANDLER] Initialize gesture handler for root view versioned.host.exp.exponent.ReactUnthemedRootView{8fb9814 V.E...... ......ID 0,0-1080,0 #1}
```

The error I see here is:

```txt
02-11 14:57:31.167  4205  4276 W ReactNativeJS: { [Error: Font.loadAsync unexpected exception: Font asset not found asset_b2e0fc821c6886fb3940f85a3320003e.ttf]
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   line: 20,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   column: 1565,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   sourceURL: '/data/user/0/com.mapeomobile/files/32.0.0/cached-bundle-experience-%40digidem%2Fmapeo2089336494-32.0.0',
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   framesToPop: 1,
02-11 14:57:31.167  4205  4276 W ReactNativeJS:   code: 'E_UNEXPECTED' }
```

The code created from scratch by the above steps is here: https://github.com/gmaclennan/mapeo-expo

**Comment by gmaclennan** [comment]
I think this is a dup of https://github.com/expo/expo/issues/3226 and the fix mentioned in 
https://github.com/expo/expo/issues/3226#issuecomment-460458263 works for me, by applying [these changes](
https://github.com/expo/expo/commit/227bdc3bd080179b7a12b04f285b578c73f7c102)

**Comment by aniltirola** [comment]
thanks @gmaclennan,

I just tested again 2 little microprojects

- one as described in the first posting of this thread
- and another with an additional image stored in `assets`; I mean the `assets` - folder in the top-level, not in `android/app/src/main/assets`
- then publish
- then eject
- then open android studio
- say 'don't remind me again' when android studio wants to do an gradle-upgrade
- then publish again twice

both projects initializied with
`expo init -t blank`


....and now everything works
.... I don't know why

--> but I am happy

**Comment by AdamJNavarro** [comment]
This should be fixed in the changes #3226 so I'm going to close this. Thanks for the report again!
